src/cli_util.rs

```diff
@@ -1848,6 +1848,28 @@ impl CliRunner {
         }
     }
 
+    /// Registers new global arguments in addition to the default ones.
+    pub fn add_global_args<A, F>(self, process_before: F) -> Self
+    where
+        A: clap::Args,
+        F: FnOnce(&mut Ui, A) -> Result<(), CommandError> + 'static,
+    {
+        let old_dispatch_fn = self.dispatch_fn;
+        let new_dispatch_fn =
+            move |ui: &mut Ui, command_helper: &CommandHelper, matches: &ArgMatches| {
+                if let Ok(custom_args) = A::from_arg_matches(matches) {
+                    process_before(ui, custom_args)?;
```

Comment:
`.unwrap()` or propagate error?

I'm not 100% sure, but I think user errors would be rejected earlier, and arguments should be parsable unlike subcommands.

Reply:
Good point. That does seem to work, and if it didn't, I agree that it would be better to not silently skip the callbacks.

I should also see if I can send another PR with some tests for `examples/`.